# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/nsGenericHTMLElement.h
# Commit: b9e87fde31f7
# Full Hash: b9e87fde31f7ca31185081dc96fa937927f1879f
# Author: sefeng <sefeng@mozilla.com>
# Date: 2020-05-12 04:04:10
# Regressor Bug: 1557303
# File Overlap Count: 1
# Description:
#   Bug 1557303 - Implement form[method="dialog"] r=smaug
#   
#   Spec: https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#submit-dialog
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73364
# ==============================================================================

diff -r e02018dfce9e -r b9e87fde31f7 dom/html/nsGenericHTMLElement.h
--- a/dom/html/nsGenericHTMLElement.h	Tue May 12 00:11:12 2020 +0000
+++ b/dom/html/nsGenericHTMLElement.h	Mon May 11 22:56:13 2020 +0000
@@ -663,6 +663,10 @@
     return aElement->IsHTMLElement(nsGkAtoms::img) && aElement->HasName();
   }
 
+  virtual inline void ResultForDialogSubmit(nsAString& aResult) {
+    GetAttr(kNameSpaceID_None, nsGkAtoms::value, aResult);
+  }
+
  protected:
   /**
    * Add/remove this element to the documents name cache